### PR TITLE
DO NOT MERGE: Add Spring Boot specific missions

### PR DESCRIPTION
**Pull request by metacosm**
This is a "controversial" PR as I'm not sure if we're allowed to do this or not. We would definitely like our users to be able to play with these community boosters, though.

### CODE REVIEW

File: metadata.json [hunk 1]
```diff
@@ -5,7 +5,9 @@
         {"id": "health-check", "name":"Health Check"},
         {"id": "rest-http", "name":"REST API Level 0"},
         {"id": "rest-http-secured", "name":"Secured"},
-        {"id": "circuit-breaker", "name":"Circuit Breaker"}
+        {"id": "circuit-breaker", "name":"Circuit Breaker"},
+        {"id": "tomcat-request-logging", "name": "Embedded Tomcat Request Logging", "description": "The main goal of this mission is to send Tomcat’s access logging to stdout, which, in the context of an application running on OpenShift, means appending access logs to the OpenShift console, thus enabling administrators to use the central logging facility to monitor Tomcat’s activity."},
+        {"id": "opentracing", "name": "Opentracing Instrumentation", "description": "This mission demonstrates how to configure and deploy Opentracing to instrument applications deployed on OpenShift."}
```
Comment (Ladicek) [inline metadata.json]: Nit: "OpenTracing"

**Comment by Ladicek** [comment]
In general, and without anticipating the results of the internal discussion whether this is allowed and in which form, I see two possible options from which we should choose:

1. Clearly mark these missions as SB-specific, e.g. by prepending the mission identifier with `sb-` or something like that.
2. Try to make them as generic as possible, if there's a possibility other runtimes will want to implement these missions too.

The `opentracing` mission is clearly using the 2nd approach. The `tomcat-request-logging` is somewhere in the middle -- it isn't very clearly marked as SB-specific (if anything, it's Tomcat-specific, but how do I know that Tomcat is SB-specific? And If JWS wanted to add boosters too, Tomcat would no longer be SB-specific!), while it isn't very general either. I can imagine it could be called`request-logging`, which is something other runtimes might as well want to implement.

My preference would probably be to use the 1st approach and switch to the 2nd the moment when a second runtime implements a particular mission. Of course that switch adds unnecessary churn, so I'm fine with using the 2nd option directly.

**Comment by gastaldi** [comment]
[test]

**Comment by gastaldi** [comment]
[test]

**Comment by valdar** [comment]
For what my opinion is worth, as "last runtime arrived onboard", I think the missions should remain as generic as possible so each runtime has a chance to implement them. Otherwise I can see this becoming very messy very quickly.

Unless the UX logic get reversed and the user is being asked to select the runtime before the missions...

**Comment by valdar** [comment]
For what my opinion is worth, as "last runtime arrived onboard", I think the missions should remain as generic as possible so each runtime has a chance to implement them. Otherwise I can see this becoming very messy very quickly.

Unless the UX logic get reversed and the user is being asked to select the runtime before the missions...

**Comment by quintesse** [comment]
Sorry @metacosm , we broke your PRs. I'm afraid you'll probably have to redo them because with the new catalog layout they will never apply cleanly. Instead of Missions being the top level folder in the catalog they are now the bottom one, with Runtimes and Versions being the new top level and second level folder respectively.

**Comment by metacosm** [comment]
Now that the UI is reversed and users select the runtime first, can we go ahead and push runtime-specific missions then?

**Comment by quintesse** [comment]
@metacosm it's definitely possible to do so. You have to remember though that all your missions will show up in the final list of missions shows in the UI. So I'm not sure if this is something that you'd want to coordinate in some way among the runtime teams.

(Imagine an extreme case where one runtime team would ad 50 new missions thereby drowning out all the missions that are supported by the other runtimes)

**Comment by metacosm** [comment]
@quintesse I haven't seen the new UI so I can't comment on this particular aspect. However, I tend to think that showing all the missions regardless of runtime availability is not a good idea. What's the use case behind this? I think RHOAR would benefit more if runtimes were free to add specific missions.

**Comment by quintesse** [comment]
@metacosm the new UI looks like this: https://redhat.invisionapp.com/share/VGF1RF15J#/screens/275859938

The reason for showing both is that you can't always predict from what PoV a user comes to the application. Are they interested in Vert.x? Or perhaps they don't care and just want to know how to create a REST application? So the new UI shows both lists of options.

If you think that's not a good idea for your team I guess it would be best to kick your concerns up the chain and/or participate in the UI design meetings.

**Comment by cmoulliard** [comment]
I think that the new UI and Catalog should allow to :  

- Filter the missions according to predefined topic such `integration`, `servicemesh`, ....
- Add new missions which are developed upstream to cover specific runtime's needs (Tomcat Logging, HTTP2, ....)

**Comment by gastaldi** [comment]
Closing for now, needs rebasing and discussion if we're accepting these starters or include them in the create an app flow

**Pull request closed without merging**